tests/schemes/test_json.py

```diff
@@ -39,22 +37,13 @@ def test_call_should_get_response():
 
     server = TChannel(name='server')
 
-    @server.register('endpoint', schemes.JSON)
-    @tornado.gen.coroutine
-    def endpoint(request, response, proxy):
+    @server.json.register
+    def endpoint(request):
 
-        headers = yield request.get_header()
-        body = yield request.get_body()
+        assert request.headers == {'req': 'headers'}
```

Comment:
This is great, a lot of methods are coroutines just to read the args we already have in memory.